File: mobile/lib/shared/widgets/luma_logo.dart
import 'dart:math' as math;

import 'package:flutter/material.dart';

import '../../core/config/theme.dart';

/// LUMA brand mark — a rounded-square tile with the aurora gradient,
/// a stylised white "L" and two sparkle accents. Rendered with a
/// [CustomPainter] so it stays sharp at any size.
class LumaLogo extends StatelessWidget {
  const LumaLogo({
    super.key,
    this.size = 64,
    this.showWordmark = false,
  });

  final double size;
  final bool showWordmark;

  @override
  Widget build(BuildContext context) {
    final mark = SizedBox.square(
      dimension: size,
      child: CustomPaint(painter: _LumaMarkPainter()),
    );

    if (!showWordmark) return mark;

    return Row(
      mainAxisSize: MainAxisSize.min,
      crossAxisAlignment: CrossAxisAlignment.center,
      children: [
        mark,
        SizedBox(width: size * 0.22),
        ShaderMask(
          shaderCallback: (r) => AppColors.primaryGradient.createShader(r),
          blendMode: BlendMode.srcIn,
          child: Text(
            'LUMA',
            style: TextStyle(
              fontSize: size * 0.74,
              fontWeight: FontWeight.w800,
              letterSpacing: -size * 0.02,
              color: Colors.white,
              height: 1.0,
            ),
          ),
        ),
      ],
    );
  }
}

class _LumaMarkPainter extends CustomPainter {
  @override
  void paint(Canvas canvas, Size size) {
    final s = size.shortestSide;
    final rect = Rect.fromLTWH(0, 0, s, s);
    final radius = Radius.circular(s * 0.234);
    final rrect = RRect.fromRectAndRadius(rect, radius);

    // Aurora gradient background.
    final bgPaint = Paint()
      ..shader = const LinearGradient(
        begin: Alignment.topLeft,
        end: Alignment.bottomRight,
        colors: [
          Color(0xFF6366F1),
          Color(0xFF8B5CF6),
          Color(0xFFEC4899),
        ],
        stops: [0.0, 0.55, 1.0],
      ).createShader(rect);
    canvas.drawRRect(rrect, bgPaint);

    // Top-right soft highlight.
    final highlightPaint = Paint()
      ..shader = RadialGradient(
        center: const Alignment(0.45, -0.55),
        radius: 0.9,
        colors: [
          Colors.white.withValues(alpha: 0.28),
          Colors.white.withValues(alpha: 0.0),
        ],
      ).createShader(rect)
      ..blendMode = BlendMode.plus;
    canvas.save();
    canvas.clipRRect(rrect);
    canvas.drawRect(rect, highlightPaint);
    canvas.restore();

    // Draw the stylised "L" — two capsule strokes that overlap.
    // Use the same 512-unit design space as the PNG assets.
    final u = s / 512.0;
    final stroke = 64.0 * u;
    final whitePaint = Paint()..color = Colors.white;

    final vertical = RRect.fromRectAndRadius(
      Rect.fromLTWH(176 * u - stroke / 2, 128 * u - stroke / 2,
          stroke, (372 - 128) * u + stroke),
      Radius.circular(stroke / 2),
    );
    canvas.drawRRect(vertical, whitePaint);

    final horizontal = RRect.fromRectAndRadius(
      Rect.fromLTWH(176 * u - stroke / 2, 372 * u - stroke / 2,
          (360 - 176) * u + stroke, stroke),
      Radius.circular(stroke / 2),
    );
    canvas.drawRRect(horizontal, whitePaint);

    // Sparkles.
    _drawSparkle(canvas, Offset(372 * u, 148 * u), 50 * u, 13 * u);
    canvas.drawCircle(Offset(372 * u, 148 * u), 9 * u, whitePaint);
    _drawSparkle(canvas, Offset(312 * u, 88 * u), 20 * u, 5 * u);
  }

  void _drawSparkle(Canvas canvas, Offset c, double rOuter, double rInner) {
    final path = Path();
    const points = 4;
    const total = points * 2;
    for (var i = 0; i < total; i++) {
      final r = i.isEven ? rOuter : rInner;
      final a = -math.pi / 2 + i * math.pi / points;
      final p = Offset(c.dx + r * math.cos(a), c.dy + r * math.sin(a));
      if (i == 0) {
        path.moveTo(p.dx, p.dy);
      } else {
        path.lineTo(p.dx, p.dy);
      }
    }
    path.close();
    canvas.drawPath(path, Paint()..color = Colors.white);
  }

  @override
  bool shouldRepaint(covariant _LumaMarkPainter oldDelegate) => false;
}

File: admin/src/components/brand/LumaLogo.js
import React from 'react';

/**
 * LUMA Aurora Mark — rounded-square aurora gradient + stylised "L" + sparkles.
 * Renders as a crisp SVG so it scales to any size.
 */
export default function LumaLogo({ size = 44, title = 'LUMA', style }) {
    const id = React.useId();
    return (
        <svg
            width={size}
            height={size}
            viewBox="0 0 512 512"
            role="img"
            aria-label={title}
            style={{ display: 'block', flexShrink: 0, ...style }}
        >
            <defs>
                <linearGradient id={`${id}-bg`} x1="0" y1="0" x2="1" y2="1">
                    <stop offset="0%" stopColor="#6366F1" />
                    <stop offset="55%" stopColor="#8B5CF6" />
                    <stop offset="100%" stopColor="#EC4899" />
                </linearGradient>
                <radialGradient id={`${id}-glow`} cx="0.72" cy="0.22" r="0.55">
                    <stop offset="0%" stopColor="#FFFFFF" stopOpacity="0.32" />
                    <stop offset="70%" stopColor="#FFFFFF" stopOpacity="0" />
                </radialGradient>
            </defs>
            <rect x="16" y="16" width="480" height="480" rx="120" ry="120" fill={`url(#${id}-bg)`} />
            <rect x="16" y="16" width="480" height="480" rx="120" ry="120" fill={`url(#${id}-glow)`} />
            {/* Stylised "L" — two capsule strokes. */}
            <rect x="144" y="96" width="64" height="308" rx="32" fill="#FFFFFF" />
            <rect x="144" y="340" width="248" height="64" rx="32" fill="#FFFFFF" />
            {/* Main sparkle. */}
            <g transform="translate(372 148)">
                <path d="M0 -50 L13 -13 L50 0 L13 13 L0 50 L-13 13 L-50 0 L-13 -13 Z" fill="#FFFFFF" />
                <circle r="9" fill="#FFFFFF" />
            </g>
            {/* Small accent sparkle. */}
            <g transform="translate(312 88)">
                <path d="M0 -20 L5 -5 L20 0 L5 5 L0 20 L-5 5 L-20 0 L-5 -5 Z" fill="#FFFFFF" />
            </g>
        </svg>
    );
}
